config/puma.rb

```diff
@@ -1,47 +1,39 @@
-# Puma can serve each request in a thread from an internal thread pool.
-# The `threads` method setting takes two numbers: a minimum and maximum.
-# Any libraries that use thread pools should be configured to match
-# the maximum value specified for Puma. Default is set to 5 threads for minimum
-# and maximum; this matches the default thread size of Active Record.
-#
-max_threads_count = ENV.fetch("RAILS_MAX_THREADS", 5)
-min_threads_count = ENV.fetch("RAILS_MIN_THREADS") { max_threads_count }
-threads min_threads_count, max_threads_count
+# This configuration file will be evaluated by Puma. The top-level methods that
+# are invoked here are part of Puma's configuration DSL. For more information
+# about methods provided by the DSL, see https://puma.io/puma/Puma/DSL.html.
 
-# Specifies the `worker_timeout` threshold that Puma will use to wait before
-# terminating a worker in development environments.
+# Puma starts a configurable number of processes (workers) and each process
+# serves each request in a thread from an internal thread pool.
+#
+# The ideal number of threads per worker depends both on how much time the
+# application spends waiting for IO operations and on how much you wish to
+# to prioritize throughput over latency.
+#
+# As a rule of thumb, increasing the number of threads will increase how much
+# traffic a given process can handle (throughput), but due to CRuby's
+# Global VM Lock (GVL) it has diminishing returns and will degrade the
+# response time (latency) of the application.
 #
-worker_timeout 3600 if ENV.fetch("RAILS_ENV", "development") == "development"
+# The default is set to 3 threads as it's deemed a decent compromise between
+# throughput and latency for the average Rails application.
+#
+# Any libraries that use a connection pool or another resource pool should
+# be configured to provide at least as many connections as the number of
+# threads. This includes Active Record's `pool` parameter in `database.yml`.
+threads_count = ENV.fetch("RAILS_MAX_THREADS", 3)
+threads threads_count, threads_count
```

Comment:
Do we need to update `config/database.yml` as well? Currently the default there is still 5.

Reply:
Good spot... I had been using RailsDiff from the previous version of Rails and not:

https://railsdiff.org/7.0.4/7.2.0#diff-60e1b9c67a0b17400eee7a22935d1cf6312f5ae9

but what's _really_ weird is that it's being reduced https://railsdiff.org/7.0.4/7.2.0#diff-60e1b9c67a0b17400eee7a22935d1cf6312f5ae9

Reply:
Bumped it down to 3 in the puma.rb

Reply:
Hmm, you're right, the default for `pool` in `database.yml` for 7.2.1 is still 5, I'm not sure why, I can't see anything about it in https://github.com/rails/rails/pull/50669 or https://github.com/rails/rails/issues/50450 (or Rails issues) 🤔 

I wonder if we should flag it up?

Reply:
sorry to clarify - the diff I checked against was between those two bump versions whereas the file hadn;t changed in our repo since 7.0.4